streamalert_cli/terraform/helpers.py

```diff
@@ -24,3 +29,33 @@ def terraform_check():
                        'your $PATH:\n'
                        '\t$ export PATH=$PATH:/usr/local/terraform/bin')
     return run_command(['terraform', 'version'], error_message=prereqs_message, quiet=True)
+
+def terraform_state_lock(action, region, table):
+    ddb_client = boto3.client('dynamodb', region_name=region)
+    ddb_tables = ddb_client.list_tables()
+    if action == 'create':
+        if table not in ddb_tables['TableNames']:
```

Comment:
I'm a fan of avoiding large nested blocks, so returning early here would make this cleaner:
```
if table in ddb_tables['TableNames']:
    return True  # table already exists
```